src/core/classes/erc-20.ts

```diff
@@ -310,12 +310,17 @@ export class Erc20<T extends TokenERC20> implements UpdateableNetwork {
    * ```
    */
   public async transferBatch(args: TokenMintInput[]) {
-    const encoded = args.map((arg) =>
-      this.contractWrapper.readContract.interface.encodeFunctionData(
+    const decimals = await this.contractWrapper.readContract.decimals();
```

Comment:
There's a helper function for this already, look for `normalizePrice`

Reply:
actually nevermind - this is fine here because we don't want to re-fetch the token info